### DISCUSSION TITLE
Detect error when adding item to cart

### DISCUSSION THREAD

**Discussion by wialy** [swellstores/community#193] [Help] [answered]
Hello everyone,

I am currently working on the cart functionality using the `swell-js` frontend library. During the implementation, I encountered an issue with the `addItem` method, which sometimes fails due to currency settings. While this is an expected behavior, the problem is that I am unable to access the error information from the SDK level even though it is included in the network request.

As a result, the addItem method always returns a `Promise<Cart>` object with the cart contents, and there seems to be no way to catch this type of error. I have reviewed the documentation and codebase, but I have not found a solution to this problem. If anyone has any insights or suggestions, I would appreciate it.

swell-js SDK version: `3.21.2`

```typescript
try {
  await addItem({
    productId,
    quantity: 1,
    variantId,
  });

  // proceed with a positive scenario
} catch {
  // handle error
}
```

**Comment by greph** [comment]
Hi @wialy , what's the error/response you're seeing when using the addItem method? I take it it's not the standard error response (with code and message), else it would trigger the catch clause.

**Comment by wialy** [comment]
Hey @greph,

Thanks for picking up the topic and sorry for my late response.

The response comes with a 200 status code, however, the payload is:

```
{"errors":{"currency":{"message":"**My Product name** is not available in currency EUR","code":"INVALID","id":"**ID**"}}}
```

It seems like the issue is not with the client SDK, maybe REST API should use different status for this kind of error.

**Reply by logeshswell** [reply, marked as answer]
@wialy Yes, you're right! It would return 200 status code and you could handle it using `response.errors` object to confirm if there's an error in placing an item in the cart.
I would also take your feedback to replace it with error status codes (4xx) while versioning the API later.
